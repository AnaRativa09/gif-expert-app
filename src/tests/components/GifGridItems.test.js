import React from 'react';
import { shallow } from 'enzyme';

import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem />', () => {

  const title = 'Mi t√≠tulo';
  const url = 'http://localhost:imagen.jpg';
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test('Debe mostrar <GifGridItem /> correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe tener un p con el title', () => {
    const p = wrapper.find('p');

    expect(p.text().trim()).toBe(title);
  });

  test('Debe tener una imagen igual al url y alt de los props', () => {
    const img = wrapper.find('img');

    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });

  test('Debe tener animate__fadeIn como className', () => {
    const div = wrapper.find('div');
    const className = div.prop('className');

    expect(className.includes('animate__fadeIn')).toBe(true);
  });
});