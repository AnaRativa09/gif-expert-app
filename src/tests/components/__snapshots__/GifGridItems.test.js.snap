// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe mostrar <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Mi título"
    src="http://localhost:imagen.jpg"
  />
  <p>
    Mi título
  </p>
</div>
`;
